src/asm/lexer.cpp

```diff
@@ -1849,7 +1849,18 @@ static Token yylex_NORMAL() {
 					}
 				}
 
-				if (token.type == T_(ID) && (lexerState->atLineStart || peek() == ':'))
+				// This is a "lexer hack"! We need it to distinguish between label definitions
+				// (which start with `LABEL`) and macro invocations (which start with `ID`).
+				//
+				// If we had one `IDENTIFIER` token, the parser would need to perform "lookahead"
+				// to determine which rule applies. An "IDENTIFIER COLON" rule would work for
+				// labels, but any macro rule would need to enter "raw" mode right after parsing
+				// the `IDENTIFIER`, which would prevent lookahead from recognizing a `COLON`.
+				//
+				// Instead, we have separate `ID` and `LABEL` tokens, lexing as a `LABEL` if a ':'
+				// character *immediately* follows the identifier. Thus, "Label:" and "mac:" are
+				// treated as label definitions, but "Label :" and "mac :" as macro invocations.
```

Comment:
I'm not super fond of this wording, because they are only treated as such at the beginning of a line. I'm worried someone familiarising themselves with rgbasm would not have that assumption, and could be confused.

Reply:
No, the whole point of this PR is that we removed that "beginning of a line" distinction. Spaceless `foo:` is always a label definition (potentially an error if `foo` is already defined, whether as a label, macro, or other); spaced `foo :` or colonless `foo` is always an identifier i.e. a macro invocation (even if `foo` is defined as a non-macro, or not defined).

Reply:
I meant “at the beginning of the line” in the “token stream” sense (where whitespace doesn't count), not the old “character stream” sense.

Reply:
Okay, I've put "at the beginning of a line", matching the language from rgbasm(5) for these two things.